test/mochitest/browser_dbg-console-eval.js

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+// Tests that clicking the DOM node button in any ObjectInspect
+// opens the Inspector panel
+
+function waitForConsolePanelChange(dbg) {
+  const { toolbox } = dbg;
+
+  return new Promise(resolve => {
+    toolbox.getPanelWhenReady("webconsole").then(() => {
+      ok(toolbox.webconsolePanel, "Console is shown.");
+      resolve(toolbox.webconsolePanel);
+    });
+  });
+}
+
+add_task(async function() {
+  const dbg = await initDebugger("doc-scripts.html", "simple2");
+
+  await selectSource(dbg, "simple2", 1);
+
+  clickElement(dbg, "codeMirrorCode");
+  await waitForElementWithSelector(dbg, ".CodeMirror-code");
+
+  // pressKey(dbg, "selectAll"); // This doesn't work properly
+  getCM(dbg).setSelection({ line: 0, ch: 0 }, { line: 8, ch: 0 });
+
+  rightClickElement(dbg, "codeMirrorCode");
+  selectContextMenuItem(dbg, "#node-menu-evaluate-in-console");
+
+  await waitForConsolePanelChange(dbg);
```

Comment:
To this point the test really only ensures that the panel switches to WebConsole; a last step is ensuring the console output is as desired, so I may need to steal a few webbconsole helper functions.  Would also like @nchevobbe's thoughts on the PR overall.

Reply:
we have 2 helpers for messages:
- you can use [findMessage](https://searchfox.org/mozilla-central/rev/78e8de968efd77ef3b30ea081bce07614cd6776b/devtools/client/webconsole/test/mochitest/head.js#240) with `waitFor` in order to wait until a given message shows up.
- [waitForMessage](https://searchfox.org/mozilla-central/rev/78e8de968efd77ef3b30ea081bce07614cd6776b/devtools/client/webconsole/test/mochitest/head.js#186), but this one is based on an event fired on the panel, so you need to have a reference to the panel before doing the evaluation, which looks tricky in this case.